presto-docs/src/main/sphinx/language/types.rst

```diff
@@ -169,19 +172,57 @@ See also :doc:`/language/timestamp`
 
     Example: ``TIME '01:02:03.456 America/Los_Angeles'``
 
+.. _timestamp-data-type:
+
 ``TIMESTAMP``
 ^^^^^^^^^^^^^
 
     Instant in time that includes the date and time of day without a time zone.
     Values of this type are parsed and rendered in the session time zone.
+    Equivalent to ``TIMESTAMP WITHOUT TIME ZONE``, ``TIMESTAMP(3)``, and
+    ``TIMESTAMP(3) WITHOUT TIME ZONE``.
 
     Example: ``TIMESTAMP '2001-08-22 03:04:05.321'``
 
+``TIMESTAMP(P) WITHOUT TIME ZONE``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+    Instant in time that includes the date and time of day without a time zone
+    with ``P`` digits of precision for the fraction of seconds.
+
+    You can use ``TIMESTAMP(P)`` as an equivalent shorter name. The default
```

Comment:
This should be the opposite. ``WITHOUT TIME ZONE`` is the equivalent longer name.

Reply:
Technically, no. The canonical name for the type in the SQL spec is “timestamp without time zone”, even though we use the short name everywhere in Presto

Reply:
You can assume my comment means "in Presto".

The the documentation should reflect the actual usage. e.g., same with `double precision`. We document what exists, not what would exist if we exactly followed the specification.

While I can see an argument for changing everything to use the longer name, that also seems like a tradeoff between readability/usability and being pedantic. (PostgreSQL seems to have chosen the latter in this case)

Reply:
I'm not suggesting we change the names to the longer form. I was just commenting on the fact that the canonical name is actually the long form and the other version is just an alias, for the benefit of others that might see this discussion in the future.